docs/computing/connecting/ssh-keys.md

```diff
@@ -0,0 +1,103 @@
+    staff. It should only be stored on the local workstation.
+
+## Generating SSH keys
+
+To find out how to generate SSH keys on your local workstation, see the
+system-specific instructions for [Unix-based systems](./ssh-unix.md) (macOS
+and Linux) and [Windows systems](./ssh-windows.md). You can come back to these
+instructions once you are prompted for a file name and storage location for
+the keys. The following instructions assume you are in the SSH key generation dialog.
+
+If you have not set up SSH keys before, feel free to accept the default name
+and location by pressing `ENTER`. However, if using the default file name
+would overwrite an existing key, you will receive a warning that looks like
+this:
+
+```
+/home/<username>/.ssh/id_ed25519 already exists. Overwrite (y/n)?
+```
+
+Generally, you do
+not want to overwrite existing keys, so enter `n`, run `ssh-keygen` again
+and enter a different file name when prompted. See also the section on
+[SSH key files with non-default name or location](#ssh-key-file-with-non-default-name-or-location).
+
+Next, you will be asked for a passphrase. Please choose a secure
+passphrase. It should be at least 8 characters long and contain numbers,
+letters and special characters.
+
+!!! warning
+    Never leave the passphrase empty when generating an SSH key pair!
+
+## Copying public key to supercomputer
+
+The recommended and easiest way to copy your public key to a CSC system is
+[through the My CSC customer portal](#adding-public-key-in-my-csc). For other
+approaches, please see the system-specific SSH instructions for Unix-based and
+Windows systems.
+
+### Adding public key in My CSC
+
+You can add your public key through the
+[My CSC customer portal](https://my.csc.fi) by following these steps:
+
+1. Log in to My CSC with your CSC or Haka/Virtu credentials.
+2. Open the dropdown menu in the top right corner and select _My Profile_.
+3. Locate _SSH PUBLIC KEYS_ section and select _Add key_. As a security
+   measure, you are asked to log in again if it has been a few minutes since
+   you last logged into the portal.
+4. Enter a _Title_ for your key pair, e.g. "my-ssh-key".
+5. Paste your **public** SSH key into the _Key_ field. The supported key types
+   are RSA 2048, ECDSA 521 and ED25519 256.
+6. Select _Add_.
+7. You should now see your new key listed under _SSH PUBLIC KEYS_. Note that
+   it might take up to one hour for your new key to become active. If it takes
+   longer than that, please
+   [contact the CSC Service Desk](../../support/contact.md).
+
+!!! info "Using compatible format"
+      Your public key must be provided in the OpenSSH format, or it is not
+      registered properly. The key must not have line breaks and there must
+      not be any spaces in the key sequence. A key in the specified format
+      looks like this:
+
+      ```
+      ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDlapOdeoxNvz/1AZFRjGAPnPj8pzzz3skI+a+yJS5b7
```

Comment:
Sometimes there is an email address in the end. Would that matter in this case? Would it work?

Reply:
I tried doing that:

`The value should consist of SSH key type and the key separated by a single space`

Reply:
I added this info in commit 1d3fc6f, so one does not have to wonder whether their key was or was not properly formatted :)